astoria/common/config.py

```diff
@@ -50,6 +51,30 @@ class Config:
 
         extra = "forbid"
 
+    @classmethod
+    def _get_config_path(cls, config_str: Optional[str] = None) -> Path:
```

Comment:
I don't think the potential bug matters this much here. The program doesn't expect the config file to be missing and there's only so much validation we can do around it. When running multiple processes on a CPU, this sort of thing is always a risk. 

I mostly separated it so that we can test it more easily. Also, the `load` function is heavily used in the test suite and I didn't want to change the behaviour and then have to modify the test suite heavily!